cmd/crates/soroban-test/tests/it/init.rs

```diff
@@ -20,68 +19,12 @@ fn init() {
         .child("Cargo.toml")
         .assert(predicate::function(|c: &str| {
             let table = toml::from_str::<toml::Table>(c).unwrap();
-            table["workspace"]["dependencies"]["soroban-sdk"].as_str()
-                == Some(&format!("{major}.0.0"))
+            let sdk_version = table["workspace"]["dependencies"]["soroban-sdk"].as_str();
+            println!("Check expected version {major}.0.0 matches template's {sdk_version:?}");
+            if is_rc {
+                sdk_version.and_then(|x| x.split('-').next()) == Some(&format!("{major}.0.0"))
+            } else {
+                sdk_version == Some(&format!("{major}.0.0"))
```

Comment:
I think we should write an assertion that the generated template is always a stable, non-rc version. See other comment.